openedx/core/djangoapps/oauth_dispatch/models.py

```diff
@@ -71,26 +71,21 @@ class ScopedApplication(AbstractApplication):
     class Meta:
         app_label = 'oauth_dispatch'
 
+    @classmethod
+    def get_scopes(cls, application):
+        return cls.objects.get(application=application).scopes
+
     def __unicode__(self):
         """
         Return a unicode representation of this object.
         """
-        return u"<ScopedApplication '{name}'>".format(
-            name=self.name
+        return u"{application_name}:{scopes}".format(
+            application_name=self.application.name,
+            scopes=self.scopes,
         )
 
-    @property
-    def authorization_filters(self):
-        """
-        Return the list of authorization filters for this application.
-        """
-        filters = [':'.join([org.provider_type, org.short_name]) for org in self.organizations.all()]
-        if self.authorization_grant_type == self.GRANT_CLIENT_CREDENTIALS:
-            filters.append(self.FILTER_USER_ME)
-        return filters
-
 
-class ScopedApplicationOrganization(models.Model):
+class ApplicationOrganizationFilter(models.Model):
```

Comment:
I suggest renaming this to simply `ApplicationOrganization` since that's all that this model provides.
The fact that we're using Organization mapping as a "filter" is a usage of this relationship.  Also, in the future, we may include further information about an app's organization that may be beyond its filter usage.

Reply:
Well we are also including the provider type which is specifically related to filters.

Reply:
Ah.  Maybe we should just call it organization type then?